azurerm/resource_arm_backup_policy_file_share.go

```diff
@@ -0,0 +1,379 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/recoveryservices/mgmt/2017-07-01/backup"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmBackupProtectionPolicyFileShare() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmBackupProtectionPolicyFileShareCreateUpdate,
+		Read:   resourceArmBackupProtectionPolicyFileShareRead,
+		Update: resourceArmBackupProtectionPolicyFileShareCreateUpdate,
+		Delete: resourceArmBackupProtectionPolicyFileShareDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[a-zA-Z][-_!a-zA-Z0-9]{2,149}$"),
+					"Backup Policy name must be 3 - 150 characters long, start with a letter, contain only letters and numbers.",
+				),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"recovery_vault_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateRecoveryServicesVaultName,
+			},
+
+			"timezone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "UTC",
+			},
+
+			"backup": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"frequency": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(backup.ScheduleRunTypeDaily),
+							}, true),
+						},
```

Comment:
My goal was to leave the schema as close to that of the `azurerm_recovery_protection_policy_vm` resource as possible since the underlying Azure resource is the same. Even though only Daily is supported now, presumably other frequencies will be supported with the service is GA and it will make less sense to have a default. My thought was this avoids having to make a breaking change in that case or be stuck with having a default value that might not make sense. Let me know if you still want me to make this change.

Reply:
👍